File: project-{topic}-plus/src/models/task.cpp
#include "task.h"
#include <ctime>
#include <iomanip>
#include <sstream>

extern std::string to_iso8601(const std::chrono::system_clock::time_point& tp); // From user.cpp

void to_json(nlohmann::json& j, const Task& t) {
    j["id"] = t.id;
    j["project_id"] = t.project_id;
    j["title"] = t.title;
    j["description"] = t.description;
    j["due_date"] = t.due_date;
    j["status"] = t.status;
    j["assigned_to_id"] = t.assigned_to_id;
    j["created_at"] = to_iso8601(t.created_at);
    j["updated_at"] = to_iso8601(t.updated_at);
}

void from_json(const nlohmann::json& j, Task& t) {
    if (j.contains("project_id")) j.at("project_id").get_to(t.project_id);
    if (j.contains("title")) j.at("title").get_to(t.title);
    if (j.contains("description")) t.description = j.at("description").get<std::optional<std::string>>();
    if (j.contains("due_date")) t.due_date = j.at("due_date").get<std::optional<std::string>>();
    if (j.contains("status")) j.at("status").get_to(t.status);
    if (j.contains("assigned_to_id")) t.assigned_to_id = j.at("assigned_to_id").get<std::optional<std::string>>();
}
```

File: project-{topic}-plus/app/models/task.py
from pydantic import BaseModel, Field
from typing import Optional
import datetime
from app.schemas.task import TaskStatus
from app.models.user import UserPublic # For assignee info
from app.models.project import ProjectListItem # For project info

# Shared properties
class TaskBase(BaseModel):
    title: str = Field(min_length=3, max_length=100)
    description: Optional[str] = Field(None, max_length=500)
    status: Optional[TaskStatus] = TaskStatus.TO_DO
    priority: Optional[int] = Field(1, ge=1, le=5) # 1 is highest, 5 is lowest
    due_date: Optional[datetime.datetime] = None
    is_completed: Optional[bool] = False

# Properties to receive via API on creation
class TaskCreate(TaskBase):
    project_id: int
    assignee_id: Optional[int] = None

# Properties to receive via API on update
class TaskUpdate(TaskBase):
    title: Optional[str] = Field(None, min_length=3, max_length=100)
    description: Optional[str] = Field(None, max_length=500)
    status: Optional[TaskStatus] = None
    priority: Optional[int] = Field(None, ge=1, le=5)
    assignee_id: Optional[int] = None # Can unset by passing None explicitly
    due_date: Optional[datetime.datetime] = None
    is_completed: Optional[bool] = None

# Properties stored in DB
class TaskInDBBase(TaskBase):
    id: int
    project_id: int
    created_at: datetime.datetime
    updated_at: datetime.datetime
    assignee_id: Optional[int] = None # Ensure assignee_id is always present here

    class Config:
        from_attributes = True

# Properties to return to API
class Task(TaskInDBBase):
    project: ProjectListItem # Embed project info
    assignee: Optional[UserPublic] = None # Embed public user info for assignee
```